File: Frontend/DiveApp/Models/UserModels.swift
import Foundation

struct ProfileSetupRequest: Encodable {
    let nickname: String
    let profileImageURL: String?
    let phoneNumber: String?
}

struct UserProfileResponse: Decodable {
    let nickname: String
    let profileImageURL: String?
    let phoneNumber: String?
    let bio: String?
}

struct CurrentUser: Decodable {
    let id: UUID
    let email: String
    let provider: AuthProvider
    let role: UserRole
    let accountStatus: AccountStatus
    let profile: UserProfileResponse?
}

File: Web/src/models/userModels.ts
import type { AccountStatus, AuthProvider, UserRole } from './enums';

export interface ProfileSetupRequest {
  nickname: string;
  profile_image_url?: string | null;
  phone_number?: string | null;
}

export interface UserProfileResponse {
  nickname: string;
  profile_image_url: string | null;
  phone_number: string | null;
  bio: string | null;
}

export interface CurrentUser {
  id: string;
  email: string;
  provider: AuthProvider;
  role: UserRole;
  account_status: AccountStatus;
  profile: UserProfileResponse | null;
}
